test/functional/tests/incremental_load/test_udev.py

```diff
@@ -0,0 +1,235 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+import random
+
+import pytest
+
+from api.cas import casadm, casadm_parser
+from api.cas.core import CoreStatus, CacheMode, CacheStatus
+from api.cas.init_config import InitConfig
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeSet, DiskTypeLowerThan, DiskType
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_udev_core_partition():
+    """
+        title: |
+          CAS udev rule execution after re-attaching partitions existing in configuration file as
+          cores.
+        description: |
+          Verify if CAS udev rule is executed when partitions existing in CAS configuration file
+          as cores are being attached.
+        pass_criteria:
+          - No kernel error
+          - Created partitions are added to core pool after attaching core drive.
+    """
+    with TestRun.step("Create four partitions on core device."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_disk = TestRun.disks["core"]
+        core_disk.create_partitions([Size(2, Unit.GibiByte) for _ in range(0, 4)])
+        core_devices = core_disk.partitions
+    with TestRun.step("Start cache and add created partitions as cores."):
+        cache = casadm.start_cache(cache_dev, force=True)
+        for i in range(0, 4):
+            cache.add_core(core_devices[i])
+    with TestRun.step("Create init config from running CAS configuration."):
+        InitConfig.create_init_config_from_running_configuration()
+    with TestRun.step("Stop cache."):
+        cache.stop()
+    with TestRun.step("Detach core disk."):
+        core_disk.unplug()
+    with TestRun.step("Plug missing core disk."):
+        core_disk.plug()
+    with TestRun.step("List cache devices and check that created partitions are present "
+                      "in core pool."):
+        for dev in core_devices:
+            check_if_dev_in_core_pool(dev)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeSet(
+    [disk_type for disk_type in DiskType if disk_type not in [DiskType.optane, DiskType.nand]]))
```

Comment:
@pytest.mark.require_disk("core", DiskTypeSet([DiskType.sata, DiskType.hdd])) would be less complicated